### PR TITLE
feat(codecov-v2): Use new Codecov API

### PR DESCRIPTION
**Pull request by snigdhas**
WOR-2723

### CODE REVIEW

File: src/sentry/api/endpoints/project_stacktrace_link.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def fetch_codecov_data(
                 ref=ref if ref else branch,
                 ref_type="sha" if ref else "branch",
                 path=path,
-                set_timeout=features.has("organizations:codecov-stacktrace-integration-v2", org),
```
Comment (snigdhas) [inline src/sentry/api/endpoints/project_stacktrace_link.py:275]: We can pass the organization instead and check the flag at a later point

File: tests/sentry/api/endpoints/test_project_stacktrace_link.py [hunk 2]
```diff
@@ -308,7 +311,7 @@ def test_munge_cocoa_worked(self, mock_integration):
         assert response.data["config"] == self.expected_configurations(self.code_mapping1)
         assert response.data["sourceUrl"] == f"{example_base_url}/{file_path}"
 
-    @mock.patch.object(ExampleIntegration, "get_stacktrace_link")
+    @patch.object(ExampleIntegration, "get_stacktrace_link")
```
Comment (armenzg) [inline tests/sentry/api/endpoints/test_project_stacktrace_link.py:314]: Up to this point, you've only swapped `mock.patch` for `patch`. 

In general, I would separate these refactoring changes into their own PR. For instance, in the next set of changes below, it requires me to pay extra attention as a reviewer.

No need to change at this point.

File: tests/sentry/api/endpoints/test_project_stacktrace_link.py [hunk 3]
```diff
@@ -399,15 +408,29 @@ def test_get_commit_sha_from_blame_failed_to_get_commit(
...
+    def test_codecov_line_coverage_success(self, mock_integration):
+        qs = urlencode(
+            {"sha": "a67ea84967ed1ec42844720d9daf77be36ff73b0", "path": "src/path/to/file.py"}
+        )
+
+        responses.add(
+            responses.GET,
+            "https://api.codecov.io/api/v2/example/getsentry/repos/sentry/report",
+            status=200,
+            match=[query_string_matcher(qs)],
```
Comment (armenzg) [inline tests/sentry/api/endpoints/test_project_stacktrace_link.py:426]: TIL

File: src/sentry/integrations/utils/codecov.py [hunk 4]
```diff
@@ -6,17 +6,21 @@
...
 CODECOV_REPORT_URL = (
     "https://api.codecov.io/api/v2/{service}/{owner_username}/repos/{repo_name}/report"
 )
+NEW_CODECOV_REPORT_URL = (
+    "https://api.codecov.io/api/v2/{service}/{owner_username}/repos/{repo_name}/file_report/{path}"
+)
 CODECOV_REPOS_URL = "https://api.codecov.io/api/v2/{service}/{owner_username}/repos"
 REF_TYPE = Literal["branch", "sha"]
 CODECOV_TIMEOUT = 2
+WALK_BACK_LIMIT = "20"
```
Comment (armenzg) [inline src/sentry/integrations/utils/codecov.py:23]: You can remove the walk back limit. We're not going to be using it.

**Review by armenzg — APPROVED** [review]
Great job! Just remove the walk back logic since we're not going to use it. It's not even mentioned in their APIs.
